rs/replicated_state/src/metadata_state.rs

```diff
@@ -2170,6 +2177,60 @@ impl
     }
 }
 
+/// Modifications to the state that require explicit tracking in order to be correctly applied
+/// by the checkpointing logic.
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub enum UnflushedCheckpointOperation {
+    /// A snapshot was deleted.
+    DeleteSnapshot(SnapshotId),
+    /// A new snapshot was created from a canister.
+    CreateSnapshot(CanisterId, SnapshotId),
+    /// A snapshot was restored to a canister.
+    RestoreSnapshot(CanisterId, SnapshotId),
```

Comment:
nit: not related to this PR, but it seems we have inconsistent naming for create/take, restore/load, and delete/remove. I'd follow the spec in the internal naming as well, i.e. take/load/delete...